libpod/options.go

```diff
@@ -2397,3 +2398,14 @@ func WithPodCPUSetCPUs(inp string) PodCreateOption {
 		return nil
 	}
 }
+
+func WithPid(inp specgen.Namespace) PodCreateOption {
+	return func(pod *Pod) error {
+		if pod.valid {
+			return define.ErrPodFinalized
+		}
+		pod.config.Pid = inp
```

Comment:
We need to require that `UsePodPID` is set in the pod configuration - we have to be sharing the pod PID namespace for this to be set.

Reply:
> We need to require that `UsePodPID` is set in the pod configuration - we have to be sharing the pod PID namespace for this to be set.

Should I require it by checking and erroring out if not set, or warn the user and set it for them?

Reply:
Good question. I was leaning towards just erroring out unless the user passes `--share` as well to specify that the PID namespace is shared, but that seems like a bad user experience. @rhatdan Should setting `--pid` automatically cause the pod to share the PID namespace?